service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/AzureStorageAccessService.java

```diff
@@ -31,43 +24,38 @@
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
 
+/**
+ * Service for Azure storage access management.
+ *
+ * <p>Rather than providing direct access to Azure storage containers, storage access is
+ * accomplished via the minting of shared access signatures (SAS).
+ */
 @Component
-public class AzureControlledStorageResourceService {
+public class AzureStorageAccessService {
 
   public record AzureSasBundle(String sasToken, String sasUrl) {}
 
   private final SamService samService;
-  private final AzureCloudContextService azureCloudContextService;
-  private final CrlService crlService;
-  private final ControlledResourceService controlledResourceService;
-  private final AzureConfiguration azureConfiguration;
   private final FeatureConfiguration features;
+  private final StorageAccountKeyProvider storageAccountKeyProvider;
 
   @Autowired
-  public AzureControlledStorageResourceService(
+  public AzureStorageAccessService(
       SamService samService,
-      AzureCloudContextService azureCloudContextService,
-      CrlService crlService,
-      ControlledResourceService controlledResourceService,
-      AzureConfiguration azureConfiguration,
+      StorageAccountKeyProvider storageAccountKeyProvider,
       FeatureConfiguration features) {
     this.samService = samService;
-    this.azureCloudContextService = azureCloudContextService;
-    this.crlService = crlService;
-    this.controlledResourceService = controlledResourceService;
-    this.azureConfiguration = azureConfiguration;
     this.features = features;
+    this.storageAccountKeyProvider = storageAccountKeyProvider;
   }
 
   private BlobContainerSasPermission getSasTokenPermissions(
-      AuthenticatedUserRequest userRequest, UUID storageContainerUuid) {
+      AuthenticatedUserRequest userRequest, UUID storageContainerUuid, String samResourceName) {
     final List<String> containerActions =
         SamRethrow.onInterrupted(
             () ->
                 samService.listResourceActions(
-                    userRequest,
-                    SamConstants.SamResource.CONTROLLED_USER_SHARED,
-                    storageContainerUuid.toString()),
+                    userRequest, samResourceName, storageContainerUuid.toString()),
```

Comment:
This is the actual fix; we are threading in the resource's Sam resource name rather than relying on the previously hardcoded value.